crates/bevy_ui/src/render/mod.rs

```diff
@@ -159,6 +159,9 @@ pub struct ExtractedUiNode {
     pub clip: Option<Rect>,
     pub flip_x: bool,
     pub flip_y: bool,
+    pub border_radius: [f32; 4],
+    pub border: [f32; 4],
+    pub invert: bool,
```

Comment:
It's a bit confusing because I tried a lot of different approaches and some things are leftover from different branches I haven't cleaned up yet. This is from an implementation that constructs borders from out of five quads. It uses this flag to mark the middle section of the border which would be drawn "inverted" so only the outside of the curved corners would be coloured. It has much better performance as long as most of the rects the UI draws aren't bordered (which is true almost all of the time) but its a bit overcomplicated and I couldn't get it to work nicely with anti-aliasing.